File: APIDemo/app/src/main/java/com/google/android/gms/example/apidemo/AdMobAdListenerFragment.java
/*
 * Copyright (C) 2015 Google, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.google.android.gms.example.apidemo;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;

/**
 * The {@link AdMobAdListenerFragment} demonstrates the use of the
 * {@link com.google.android.gms.ads.AdListener} class.
 */
public class AdMobAdListenerFragment extends Fragment {

    private AdView mAdView;

    public AdMobAdListenerFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_admob_ad_listener, container, false);
        return rootView;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        mAdView = (AdView) getView().findViewById(R.id.listener_av_main);

        mAdView.setAdListener(new AdListener() {
            private void showToast(String message) {
                View view = getView();
                if (view != null) {
                    Toast.makeText(getView().getContext(), message, Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onAdLoaded() {
                showToast("Ad loaded.");
            }

            @Override
            public void onAdFailedToLoad(int errorCode) {
                showToast(String.format("Ad failed to load with error code %d.", errorCode));
            }

            @Override
            public void onAdOpened() {
                showToast("Ad opened.");
            }

            @Override
            public void onAdClosed() {
                showToast("Ad closed.");
            }

            @Override
            public void onAdLeftApplication() {
                showToast("Ad left application.");
            }
        });

        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);
    }
}

File: kotlin/advanced/APIDemo/app/src/main/java/com/google/android/gms/example/apidemo/AdMobAdListenerFragment.kt
package com.google.android.gms.example.apidemo

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.google.android.gms.ads.AdListener
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.LoadAdError
import com.google.android.gms.example.apidemo.databinding.FragmentAdmobAdListenerBinding

/**
 * The [AdMobAdListenerFragment] demonstrates the use of the [com.google.android.gms.ads.AdListener]
 * class.
 */
class AdMobAdListenerFragment : Fragment() {

  private lateinit var fragmentBinding: FragmentAdmobAdListenerBinding

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?,
  ): View? {
    fragmentBinding = FragmentAdmobAdListenerBinding.inflate(inflater)
    return fragmentBinding.root
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)

    fragmentBinding.listenerAdView.adListener =
      object : AdListener() {
        private fun showToast(message: String) {
          val context = this@AdMobAdListenerFragment.view?.context
          if (context != null) {
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
          }
        }

        override fun onAdLoaded() {
          showToast("Ad loaded.")
        }

        override fun onAdFailedToLoad(loadAdError: LoadAdError) {
          val error =
            "domain: ${loadAdError.domain}, code: ${loadAdError.code}, " +
              "message: ${loadAdError.message}"
          showToast("Ad failed to load with error $error.")
        }

        override fun onAdOpened() {
          showToast("Ad opened.")
        }

        override fun onAdClosed() {
          showToast("Ad closed.")
        }
      }

    val adRequest = AdRequest.Builder().build()
    fragmentBinding.listenerAdView.loadAd(adRequest)
  }
}
